### PR TITLE
GCS: fix the WithoutAuthentication option

### PR DESCRIPTION
**Pull request by haoxins**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

## What problem(s) does this PR solve?

When you run the application in the GCP ENVs, such as GKE or GCE.
The GCS client will try to get credentials underlying the ENVs.

See https://cloud.google.com/kubernetes-engine/docs/concepts/workload-identity

Unless the user specifies that, we shouldn't set the `WithoutAuthentication` option.

### CODE REVIEW

File: pkg/source/gcs.go [hunk 1]
```diff
@@ -13,11 +13,12 @@ var _ Source = (*gcsSource)(nil)
...
 	GCSConfig struct {
-		Endpoint        string `yaml:"endpoint,omitempty"`
-		CredentialsFile string `yaml:"credentialsFile,omitempty"`
-		CredentialsJSON string `yaml:"credentialsJSON,omitempty"`
-		Bucket          string `yaml:"bucket,omitempty"`
-		Key             string `yaml:"key,omitempty"`
+		Endpoint              string `yaml:"endpoint,omitempty"`
+		CredentialsFile       string `yaml:"credentialsFile,omitempty"`
+		CredentialsJSON       string `yaml:"credentialsJSON,omitempty"`
+		WithoutAuthentication bool   `yaml:"withoutAuthentication,omitempty"`
```
Comment (veezhang) [inline pkg/source/gcs.go:19]: The current implementation is that if `CredentialsFile` and `CredentialsJSON` are not configured, then it is considered that there is no authentication. Is there any problem with this?

**Comment by veezhang** [comment]
@haoxins Thanks!
Can you fix the testing ?
BTW, do you know which env has the effect?

**Comment by haoxins** [comment]
I need to find a solution to fix the test.
The test failed because it is not running on the GKE/GCE environment.

**Comment by haoxins** [comment]
> @haoxins Setting `WithoutAuthentication` to `true` may solve the problem.

fixed

**Comment by veezhang** [comment]
@haoxins Thanks a lot!

**Review by yixinglu — APPROVED** [review]

**Pull request merged**